src/main/java/com/epam/test_generator/services/DefaultStepSuggestionService.java

```diff
@@ -0,0 +1,118 @@
+                                                       Integer pageSize) {
+        if (isPageNumberAndPageSizeAndStepTypeNotNull(stepType, pageNumber, pageSize)) {
+            return getStepsSuggestionsByType(stepType, pageNumber, pageSize);
+        }
+        if (isPageNumberAndPageSizeNotNull(pageNumber, pageSize)) {
+            return getStepsSuggestions(pageNumber, pageSize);
+        }
+        if (stepType != null) {
+            return getStepsSuggestionsByType(stepType);
+        }
+        return getStepsSuggestions();
+    }
+
+    private List<StepSuggestionDTO> getStepsSuggestions() {
+        return defaultStepSuggestionTransformer.toDtoList(defaultStepSuggestionDAO.findAll());
+    }
+
+    private List<StepSuggestionDTO> getStepsSuggestions(int pageNumber, int pageSize) {
+        Pageable request = new PageRequest(pageNumber - 1, pageSize, Sort.Direction.ASC, "id");
+        return defaultStepSuggestionTransformer.toDtoList(defaultStepSuggestionDAO.findAll(request).getContent());
+    }
+
+    private boolean isPageNumberAndPageSizeNotNull(Integer pageNumber, Integer pageSize) {
+        return pageNumber != null && pageSize != null;
+    }
+
+    private boolean isPageNumberAndPageSizeAndStepTypeNotNull(StepType stepType, Integer pageNumber,
+                                                              Integer pageSize) {
+        return isPageNumberAndPageSizeNotNull(pageNumber, pageSize) && stepType != null;
+    }
+
+    public StepSuggestionDTO getStepsSuggestion(long stepSuggestionId) {
+        DefaultStepSuggestion defaultStepSuggestion = defaultStepSuggestionDAO.findOne(stepSuggestionId);
+        checkNotNull(defaultStepSuggestion);
+
+        return defaultStepSuggestionTransformer.toDto(defaultStepSuggestion);
+    }
+
+    private List<StepSuggestionDTO> getStepsSuggestionsByType(StepType stepType, int pageNumber, int pageSize) {
+        Pageable request = new PageRequest(pageNumber - 1, pageSize, Sort.Direction.ASC, "id");
+
+        return defaultStepSuggestionTransformer.toDtoList(
+                defaultStepSuggestionDAO.findAll(request).getContent().stream()
+                        .filter(s -> s.getType() == stepType)
+                        .collect(Collectors.toList()));
+    }
+
+    public List<StepSuggestionDTO> getStepsSuggestionsByType(StepType stepType) {
+        return defaultStepSuggestionTransformer.toDtoList(
+                defaultStepSuggestionDAO.findAll().stream()
+                        .filter(s -> s.getType() == stepType)
+                        .collect(Collectors.toList()));
+    }
+
+    /**
+     * Find all steps suggestions by string ignoring case
+     *
+     * @param searchString - string for search
+     * @param limit - count of results
+     * @return list with found steps suggestions
+     */
+    public List<StepSuggestionDTO> findStepsSuggestions(String searchString, int limit) {
+        if(searchString.isEmpty()) {
```

Comment:
This validation should be performed in controller.

Reply:
Done. 
I removed validation logic for searchString, because there is "required" parameter in Swagger annotation. Is it ok?